### PR TITLE
Parse RADIUS request attributes during Redis cache extraction

### LINKED ISSUE #8358: Radius Audit Log UI displays WLAN-AKM-Suite, WLAN-Group-Cipher, and WLAN-Pairwise-Cipher as float/double values and TLS cert dates are unprocessed
**Describe the bug**
The "Radius Audit Log" UI displays WLAN-AKM-Suite, WLAN-Group-Cipher, and WLAN-Pairwise-Cipher as float/double values. Also TLS cert valid from/to dates aren't processed. These issues exists in PF 14.0 and 13.2, I can't confirm if it exists in other versions.

**To Reproduce**
1. Generate a RADIUS audit log entry for an EAP-TLS client authentication.
2. View the WLAN-AKM-Suite, WLAN-Group-Cipher, WLAN-Pairwise-Cipher, TLS-Cert-Expiration, TLS-Cert-Valid-Since, TLS-Client-Cert-Expiration, and TLS-Client-Cert-Valid-Since fields

**Expected behavior**
The WLAN-AKM-Suite, WLAN-Group-Cipher, and WLAN-Pairwise-Cipher should not be displayed as float/double values. Either display raw hex or parse to display the decoded fields. The TLS-Cert-Expiration, TLS-Cert-Valid-Since, TLS-Client-Cert-Expiration, and TLS-Client-Cert-Valid-Since should display a date. There are other fields that are displayed as hex, such as EAP-Message, State, PacketFence-KeyBalanced and others. If those can easily be parsed, they should be. Hex is fine if the alternatives aren't simple or are resource prohibitive.

### PR DESCRIPTION
**Pull request by E-Thang**
# Description
Parse RADIUS attributes  WLAN-AKM-Suite, WLAN-Group-Cipher, WLAN-Pairwise-Cipher, TLS-Cert-Expiration, TLS-Cert-Valid-Since, TLS-Client-Cert-Expiration, and TLS-Client-Cert-Valid-Since into human readable strings.

Addresses issue where WLAN AKM and Cipher suite information integer value is displayed as a float64.

Note: Parsing could be done elsewhere, however, the pfcron flush task is one of the last locations where the RADIUS information is still in JSON and not a formatted string. It's relatively easy to manipulate now. Having said that, from a resource utilization perspective it may make more sense to perform this task as a particular RADIUS log entry is requested from the Admin UI.

Also, there may be additional attributes that would be nice to have parsed. I can add more if desired. The EAP-Message is perhaps a reasonable addition.

# Impacts
Usability of information presented in Admin UI RADIUS audit log

# Issue
fixes #8358 

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add OpenAPI specification
- [ ] (If required) Add unit tests
- [ ] Add acceptance tests (TestLink)

## Enhancements
Added parsing of WLAN-AKM-Suite, WLAN-Group-Cipher, WLAN-Pairwise-Cipher, TLS-Cert-Expiration, TLS-Cert-Valid-Since, TLS-Client-Cert-Expiration, and TLS-Client-Cert-Valid-Since attributes.

WLAN AKM and Cipher Suite attributes are decoded and filled with the name of the AKM or suite used. Dates are presented in a human-readable format with 4-digit year.

Existing behavior with WLAN attributes is that the integers are erroneously unmarshaled as Float64 by Golang json.Unmarshal. Date's are presented as a string with 2-digit year and no field delimiters or spacing ("241025235210Z").

Example of change as viewed from Admin UI:
```
TLS-Cert-Expiration = "2039-04-10 17:00:00",
TLS-Cert-Valid-Since = "2019-04-10 16:51:00",
TLS-Client-Cert-Expiration = "2029-09-27 21:14:28",
TLS-Client-Cert-Valid-Since = "2024-09-27 21:04:00",
WLAN-AKM-Suite = "FT over 802.1X",
WLAN-Group-Cipher = "CCMP-128",
WLAN-Pairwise-Cipher = "CCMP-128"
```
image

### CODE REVIEW

File: go/cron/flush_radius_audit_log_job.go [hunk 1]
```diff
@@ -340,3 +341,169 @@ func interfaceToStr(i interface{}, defaultStr string) string {
...
+			return "FT over FILS-SHA384"
+		case OWE_transition_mode:
+			return "OWE transition mode"
+		default:
+			return fmt.Sprintf("Unknown or Reserved AKM suite (Value: %d)", a)
+	}
+}
+
+func mapAKMSuite(akmSuiteInt int) string {
+	akmSuiteSelector: = akmSuiteInt & 0x0000000F
```
Comment (fdurand) [inline go/cron/flush_radius_audit_log_job.go]: typo here should be := and not : =

File: go/cron/flush_radius_audit_log_job.go [hunk 2]
```diff
@@ -340,3 +341,169 @@ func interfaceToStr(i interface{}, defaultStr string) string {
...
+	return AKMSuite(akmSuiteSelector).String()
+}
+
+func mapCipherSuite(cipherSuiteInt int) string {
+	cipherSuiteSelector: = cipherSuiteInt & 0x0000000F
```
Comment (fdurand) [inline go/cron/flush_radius_audit_log_job.go]: typo here should be := and not : =

**Review by fdurand — CHANGES_REQUESTED** [review]
Except the 2 typos the code works.

File: go/cron/flush_radius_audit_log_job.go (hunk 1 above)
Comment (E-Thang) [reply go/cron/flush_radius_audit_log_job.go]: Fixed and pushed changes

File: go/cron/flush_radius_audit_log_job.go (hunk 2 above)
Comment (E-Thang) [reply go/cron/flush_radius_audit_log_job.go]: Fixed and pushed changes

**Comment by E-Thang** [comment]
Can you think of other attributes that it would be handy to include parsing of?

**Comment by fdurand** [comment]
> Can you think of other attributes that it would be handy to include parsing of?

Not really. At least we can keep like that and we will see if we need to add more.

**Pull request merged**